### PR TITLE
[enhancement](nereids) Support microseconds_add and microseconds_sub for nereids

### PR DESCRIPTION
**Pull request by zzzzzzzs**
## Proposed changes

Support microseconds_add and microseconds_sub for nereids

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/MicrosecondsAdd.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * ScalarFunction 'microseconds_add'.
+ */
+public class MicrosecondsAdd extends ScalarFunction
+        implements BinaryExpression, ExplicitlyCastableSignature, PropagateNullableOnDateLikeV2Args {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/MicrosecondsAdd.java:39]: if args only have datev2/datetimev2, not contains date/datetime, use PropagateNullable directly is ok

Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/MicrosecondsAdd.java]: done

**Pull request closed without merging**